internal/service/dns_config/api_auth_nsg_resource_test.go

```diff
@@ -0,0 +1,442 @@
+package dns_config_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/dns_config"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+//TODO: add tests
+// The following require additional resource/data source objects to be supported.
+// - internal_secondaries
+// - TSIG keys
+
+func TestAccAuthNsgResource_basic(t *testing.T) {
+	var resourceName = "bloxone_dns_auth_nsg.test"
+	var v dns_config.ConfigAuthNSG
+	name := acctest.RandomNameWithPrefix("auth-nsg")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAuthNsgBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAuthNsgExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAuthNsgResource_disappears(t *testing.T) {
+	resourceName := "bloxone_dns_auth_nsg.test"
+	var v dns_config.ConfigAuthNSG
+	name := acctest.RandomNameWithPrefix("auth-nsg")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckAuthNsgDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAuthNsgBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAuthNsgExists(context.Background(), resourceName, &v),
+					testAccCheckAuthNsgDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAuthNsgResource_Comment(t *testing.T) {
+	var resourceName = "bloxone_dns_auth_nsg.test_comment"
+	var v dns_config.ConfigAuthNSG
+	name := acctest.RandomNameWithPrefix("auth-nsg")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAuthNsgComment(name, "COMMENT_REPLACE_ME"),
```

Comment:
While this is ok `"COMMENT_REPLACE_ME"`. I would suggest we have a proper comment here :). I will update this in my PR